### PR TITLE
php7: prepare to merge with grpc

### PR DESCRIPTION
**Pull request by stanley-cheung**
Hi, Thanks again for the contribution. Let's get this ready to be merged back into the grpc repo. Here's what I propose:
- Please review this PR, and merge into your `thinkerou/grpc-php7` repo
- ~~then go to your `thinkerou/grpc` fork, create a new directory `src/php/ext/grpc/php7` and copy all the files from this repo's `ext/grpc` directory there. So for example: `thinkerou/grpc-php7/ext/grpc/call.c` goes to `thinkerou/grpc/src/php/ext/grpc/php7/call.c` etc.~~
- Then go to your `thinkerou/grpc` fork, for each file in `grpc/src/php/ext/grpc` directory, wrap the existing code like this:

```
#if PHP_MAJOR_VERSION < 7
<existing php5 code>
#else
<new php7 code>
#endif
```
- create a PR to the `grpc/grpc` main repo and assign the PR to me. 
- then we can both contribute to that PR to get this ready for the main `grpc/grpc` repo.

For this PR:
- I mostly just cleaned up the code a bit. Removed commented out code. Remove trailing whitespaces. Make sure all lines <80 characters etc.
- I renamed some variables
- I removed the `FAST_ZPP` stuff for now. It's not mandatory and I'd like to keep this as simple as possible for now. We can add these back later
- Removed a couple remaining `TSRMLS_CC` macro
- May have fixed a couple of issues along the way

Please let me know if you have any questions, concerns. Thanks!

### CODE REVIEW

**Comment by stanley-cheung** [comment]
On second thought, we probably should not create a new `grpc/src/php/ext/grpc/php7` directory, because that will create the need for php5 and php7 users to install a different extension (by version). 

Instead, we should keep the current file structure and wrap the different code around blocks like this

```
#if PHP_MAJOR_VERSION < 7
<existing php5 code>
#else
<new php7 code>
#endif
```

Some of the files have minimal change between php5 and php7, like `byte_buffer.c` so there is probably no need for those blocks. But for files like `call.c` that have lots of change, I am OK with wrapping almost the entire existing file into a code block above like <existing php5 code>.

Can you do that when you submit the PR to the `grpc/grpc` repo please? Thanks

**Comment by thinkerou** [comment]
Hi, @stanley-cheung I think that we should not use the method:

```
#if PHP_MAJOR_VERSION < 7
    <existing php5 code>
#else
    <new php7 code>
#endif
```

The method will have many determine macro, but the method may be make code clear.

Now I attempt to write wrapper `php7_wrapper.h`, like the follow:

```
#ifndef PHP7_WRAPPER_GRPC_H
#define PHP7_WRAPPER_GRPC_H

#if PHP_MAJOR_VERSION < 7

#define GRPC_WRAP_OBJECT_START(name) \
    typedef struct name { \
      zend_object std;

#define GRPC_WRAP_OBJECT_END(name) } name;

#else

#define GRPC_WRAP_OBJECT_START(name) typedef struct name {

#define GRPC_WRAP_OBJECT_END(name) \
    zend_object std; \
  } name;

#endif
```

The method will use much less than before method, but it will make code difficult understand.

what do you think?

File: ext/grpc/config.m4 [hunk 1]
```diff
@@ -80,7 +76,7 @@ if test "$PHP_GRPC" != "no"; then
 
   PHP_NEW_EXTENSION(grpc, byte_buffer.c call.c call_credentials.c channel.c \
     channel_credentials.c completion_queue.c timeval.c server.c \
-    server_credentials.c php_grpc.c, $ext_shared, , -Wall -Werror -std=c11)
+    server_credentials.c php_grpc.c, $ext_shared, , -Wall -Werror -Wno-maybe-uninitialized -std=c11)
```
Comment (thinkerou) [inline ext/grpc/config.m4:79]: Have not the param on mac, but have `-Wno-uninitialized`.

**Comment by thinkerou** [comment]
I have merged your commit, and I commit some codes which update unit test case for deleting `setErrorHandler` and fix crash for no init optional param.

But, you need to sure `-Wno-maybe-uniitialized` param.

File: ext/grpc/config.m4 (hunk 1 above)
Comment (stanley-cheung) [inline ext/grpc/config.m4:79]: Yea this is probably temporary on my debian instance. We can worry about this later.

**Comment by stanley-cheung** [comment]
I like the `php7_wrapper.h` idea. Let's do that. I have seen some other extension developer do that. But there are still plenty of forked code in `.c` files. I am OK with some amount of `#if PHP_MAJOR_VERSION < 7`. As long as duplicate code are in the same file, it's slightly more easily manageable.

**Comment by thinkerou** [comment]
OK, I have committed  part codes on `grpc/grpc`, please review, thanks.

**Pull request merged**